framework/src/transfers/MultiAppGeneralFieldTransfer.C

```diff
@@ -430,20 +624,37 @@ MultiAppGeneralFieldTransfer::cacheOutgoingPointInfo(const Point point,
                                                      ProcessorToPointVec & outgoing_points)
 {
   std::set<processor_id_type> processors;
-  // Try to find which processors
+  // Find which processors will receive point data so they can send back value data
+  // The list can be larger than needed, depending on the heuristic / algorithm used to make
+  // the call on whether a processor (and the apps it runs) should be involved
   processors.clear();
   locatePointReceivers(point, processors);
 
-  // We need to send these data to these processors
-  for (auto pid : processors)
+  // We need to send this location data to these processors so they can send back values
+  for (const auto pid : processors)
   {
-    outgoing_points[pid].push_back(point);
-    // Store point information
+    // Select which from_mesh_division the source data must come from for this point
+    unsigned int required_source_division = 0;
+    if (_from_mesh_division_behavior == MeshDivisionTransferUse::MATCH_SUBAPP_INDEX)
+      required_source_division = getGlobalTargetAppIndex(problem_id);
+    else if (_from_mesh_division_behavior == MeshDivisionTransferUse::MATCH_DIVISION_INDEX ||
+             _to_mesh_division_behavior == MeshDivisionTransferUse::MATCH_DIVISION_INDEX ||
+             _to_mesh_division_behavior == MeshDivisionTransferUse::MATCH_SUBAPP_INDEX)
+      required_source_division = _to_mesh_divisions[problem_id]->divisionIndex(
+          _to_transforms[getGlobalTargetAppIndex(problem_id)]->mapBack(point));
```

Comment:
yes there's the option to just use spatial restriction

Reply:
in that case, that we make it with required_source_divison = 0 because we are just using it as a spatial restriction, then `required_source_divison` will not be used later on so it does not matter its value